src/components/vm/disks/diskAdd.jsx

```diff
@@ -509,11 +510,38 @@ export class AddDiskModalBody extends React.Component {
             break;
         }
         case 'file': {
+            this.setState({ file: value });
+
             if (value.endsWith(".iso")) {
                 // use onValueChange instead of setState in order to perform subsequent state change logic
                 this.onValueChanged("device", "cdrom");
             }
-            this.setState({ file: value });
+
+            if (value) {
+                // Limit process time to 2 seconds and ram to 100 MB
+                cockpit.spawn(["/bin/bash", "-c", "ulimit -t 2 -v 100000; echo -n " + value + " | xargs -0n1 qemu-img info --output json "], { err: "message" })
```

Comment:
No, that doesn't help -- it just moves the shell escaping madness to a slightly different place in the command line. You really need to `.input(value)` here to feed the name through stdin.